core/device.cpp

```diff
@@ -252,3 +252,54 @@ extern "C" void create_fingerprint_node(struct fingerprint_table *table, uint32_
 		table->fingerprints.insert(it, fpr);
 	}
 }
+
+extern "C" void create_fingerprint_node_from_hex(struct fingerprint_table *table, uint32_t model, uint32_t serial,
+						const char *hex_data, uint32_t fdeviceid, uint32_t fdiveid)
+{
+	QByteArray raw = QByteArray::fromHex(hex_data);
+	create_fingerprint_node(table, model, serial,
+				(const unsigned char *)raw.constData(), raw.size(), fdeviceid, fdiveid);
+}
+
+extern "C" int nr_fingerprints(struct fingerprint_table *table)
+{
+	return table->fingerprints.size();
+}
+
+extern "C" uint32_t fp_get_model(struct fingerprint_table *table, int i)
+{
+	if (!table || i < 0 || i >= table->fingerprints.size())
+		return 0;
+	return table->fingerprints[i].model;
+}
+
+extern "C" uint32_t fp_get_serial(struct fingerprint_table *table, int i)
```

Comment:
I'm not a fan of these verbose accessors for each field. No big deal, though.

Reply:
yeah, me neither... but with the C/C++ interface here... I couldn't come up with a better way that wouldn't seriously break our layering...